feat: 新增模型前缀删除功能

Silentely opened the pull request:
## 概述
本次PR新增了模型前缀删除功能，允许用户配置前缀列表，自动从模型名称中移除指定前缀，提升模型管理的灵活性。

## 主要改动

### 🔧 核心新功能
- **删除模型前缀功能**：支持配置模型前缀列表，自动从模型名称中移除指定前缀
  - 例如：`openai/gpt-4`、`deepseek/deepseek-chat` 等模型可自动去除 `openai/`、`deepseek/` 前缀
  - 支持多个前缀配置，一键批量处理
  - 在模型映射对话框中提供便捷的管理界面
  - 支持Enter键快速添加，重复前缀自动检测

### 🔧 技术实现
- **后端支持**：完整的Go后端实现
  - Ent Schema 支持 `removeModelPrefixes` 字段
  - Transformer 处理逻辑，自动应用前缀删除
  - GraphQL 接口更新
- **前端实现**：
  - React 组件集成，支持动态添加/删除前缀
  - Zod schema 校验
  - 国际化支持（中英文）
- **数据处理**：
  - 模型名称自动处理
  - 支持复杂模型路径解析

## 使用场景
1. **多渠道模型统一**：不同AI提供商使用相同基础模型，但前缀不同
2. **模型名称简化**：将 `openai/gpt-4` 简化为 `gpt-4` 便于管理
3. **批量处理**：一次性处理多个带前缀的模型

## 测试建议
1. 测试添加/删除前缀功能
2. 验证模型名称前缀自动删除效果
3. 测试重复前缀检测和警告提示
4. 检查Enter键快速添加功能
5. 验证国际化文本显示

## 兼容性
- 向后兼容，不影响现有渠道配置
- 新增字段为可选，现有数据不受影响

Silentely committed:
feat: 为渠道添加模型前缀删除功能和改进UI交互

- 新增支持配置自动移除的模型前缀功能，简化模型名称匹配
- 改进渠道设置对话框的UI交互，优化模型显示和操作流程
- 限制主界面显示支持模型为3个，需查看全部时通过面板访问
- 在模型映射面板中移动"清除全部"按钮，移除主界面的冗余操作
- 优化自动完成选择器的搜索体验，打开时清空搜索显示所有选项
- 更新前后端数据模型和GraphQL模式以支持新功能
- 添加中英文本地化文本支持新的用户界面元素

Silentely committed:
refactor: 重构模型前缀添加逻辑，提取为可复用函数

- 将重复的前缀添加逻辑提取为 `handleAddPrefix` 函数，使用 `useCallback` 进行优化
- 统一处理表单操作，避免直接使用 `form.watch` 和 `form.setValue` 的重复代码
- 修复 `onKeyPress` 事件为 `onKeyDown` 以正确处理 Enter 键提交
- 初始化表单时正确设置 `removeModelPrefixes` 字段，确保数据一致性
- 简化 JSX 中的内联函数，提高代码可读性和维护性

looplj commented on internal/objects/channel.go:
```diff
@@ -41,6 +41,10 @@ type ChannelSettings struct {
 	ModelMappings []ModelMapping `json:"modelMappings"`
 
+	// RemoveModelPrefixes configures prefixes to automatically remove from model names.
+	// e.g. ["openai", "deepseek"] will transform "openai/gpt-4" to "gpt-4" and "deepseek/deepseek-chat" to "deepseek-chat".
```
名字叫 RemoveModelPrefixes 比较合适一些

looplj replied:
说错了，应该叫 AutoRemoveModelPrefixes

looplj replied:
说错了，应该叫 AutoRemoveModelPrefixes

looplj commented on internal/server/biz/channel_llm.go:
```diff
@@ -46,6 +61,8 @@ func (c Channel) resolvePrefixedModel(model string) (string, bool) {
 }
 
 func (c Channel) IsModelSupported(model string) bool {
+	model = c.removeModelPrefixes(model)
```
应该放到后面 resolvePrefixedModel 后面

looplj commented on internal/server/biz/channel_llm.go:
```diff
@@ -79,6 +96,8 @@ func (c *Channel) CustomizeExecutor(executor pipeline.Executor) pipeline.Executo
 }
 
 func (c Channel) ChooseModel(model string) (string, error) {
+	model = c.removeModelPrefixes(model)
```
同样应该放到后面

looplj commented on internal/server/biz/channel_llm.go:
```diff
@@ -27,6 +27,21 @@ import (
 	"github.com/looplj/axonhub/internal/pkg/httpclient"
 )
 
+func (c Channel) removeModelPrefixes(model string) string {
```
方法名改为 resolveAutoRemovedPrefixes

looplj replied:
需要添加测试用例

Silentely committed:
fix: 调整模型前缀移除逻辑的顺序以提高匹配准确性

- 将自动移除前缀逻辑从模型匹配流程的起始位置移至更合适的位置
- 在检查原始模型、前缀解析模型之后，再应用自动移除前缀功能
- 确保模型映射功能优先于自动前缀移除，保持向后兼容性
- 移除前端中未使用的`handleAddPrefix`函数以清理代码
- 添加全面的单元测试验证`resolveAutoRemovedPrefixes`方法的正确性

Silentely committed:
Merge branch 'unstable' into feat/remove-prefixes

Silentely commented:
提交了~

looplj commented on internal/server/biz/channel_llm.go:
```diff
@@ -58,6 +73,12 @@ func (c Channel) IsModelSupported(model string) bool {
 
+	model = c.resolveAutoRemovedPrefixes(model)
+
+	if slices.Contains(c.SupportedModels, model) {
```
这一段重复了。

looplj commented:
字段名字叫 RemoveModelPrefixes 比较合适一些

looplj commented:
本地需要测试一下，刚才的 PR 本地有错误，因为前端代码不好用 CI 去测试，需要人工测试一下。

looplj commented:
先确认一下这个功能的逻辑和 extra model prefix 不一样吧。

extra model prefix ： 渠道有 deepseek-chat，prefix 是 deepseek ，然后可以通过  deepseek/depseek-chat 访问，最终请求的是：depseek-chat

auto remove mode prefix 是：渠道有 deepseek/depseek-chat ，删除 prefix deepseek，然后可以通过  depseek-chat  访问到 ?最终请求的是：deepseek/depseek-chat ？


确定是不是这个业务逻辑，是的话，现在的代码应该有点问题，可以在 test cases 里面补充这些业务逻辑，要测试 IsSupported 和 Choose models ，然后修复

Silentely committed:
refactor: 重构模型前缀移除功能并改进前端交互

- 将后端 `resolveAutoRemovedPrefixes` 方法重命名为 `RemoveModelPrefixes` 以提高可读性
- 修复前端移除前缀标签的点击事件，使用 `form.getValues` 并添加表单验证触发
- 恢复模型映射对话框的描述文本显示
- 更新相关测试用例以匹配新的方法名称

Silentely committed:
feat: 使模型前缀移除功能支持双向映射

- 修改 `RemoveModelPrefixes` 逻辑，使其支持双向模型名称匹配
  * 当请求包含前缀时，若去掉前缀后的模型在支持的模型列表中，则返回去掉前缀的模型
  * 当请求不包含前缀时，若添加前缀后的模型在支持的模型列表中，则返回添加前缀的模型
- 更新前端中英文本地化文件，调整功能描述以反映新的双向行为
- 添加单元测试验证前缀移除功能的对称性
- 补充现有测试用例中缺失的 `SupportedModels` 字段设置

Silentely commented:
本地测试了，现在是配置前缀为 'deepseek-ai'，模型 'deepseek-ai/deepseek-chat' 将变为 'deepseek-chat'， 'deepseek-ai/deepseek-chat' 和 'deepseek-chat' 两个都可以调用
不行了，顶不住了，睡了
<img width="781" height="244" alt="image" src="https://github.com/user-attachments/assets/2375b08b-3132-428e-8f99-b0912451c61a" />
<img width="228" height="123" alt="image" src="https://github.com/user-attachments/assets/ae566007-9fe4-457b-a0ef-56f1bd707acc" />
<img width="882" height="527" alt="image" src="https://github.com/user-attachments/assets/a4532d82-285b-4159-87fb-3ea3d0e2e2af" />
<img width="808" height="93" alt="image" src="https://github.com/user-attachments/assets/13e1ae30-3745-4928-93da-b74994ce2dd2" />

Silentely committed:
Merge branch 'unstable' into feat/remove-prefixes

looplj commented:
感谢 PR，代码我先合了，后面我再测试调整下。
